indico/modules/rb_new/client/js/components/TimeRangePicker.jsx

```diff
@@ -0,0 +1,198 @@
+
+
+const _serializeTime = time => (time ? time.format('HH:mm') : null);
+
+function _humanizeDuration(duration) {
+    const hours = duration.hours();
+    const minutes = duration.minutes();
+    if (hours === 1 && minutes === 0) {
+        return Translate.string('1 hour');
+    } else if (hours !== 0) {
+        return Translate.string('{time} hours', {time: hours + (minutes / 60)});
+    } else {
+        return Translate.string('{time} minutes', {time: minutes});
+    }
+}
+
+export default class TimeRangePicker extends React.Component {
+    static propTypes = {
+        startTime: propTypes.object,
+        endTime: propTypes.object,
+        onChange: propTypes.func.isRequired,
+    };
+
+    static defaultProps = {
+        startTime: moment().startOf('hour').add(1, 'h'),
+        endTime: moment().startOf('hour').add(2, 'h')
+    };
+
+    constructor(props) {
+        super(props);
+
+        const {startTime, endTime} = this.props;
+        const startOptions = this.generateStartTimeOptions();
+        const endOptions = this.generateEndTimeOptions(startTime);
+        const duration = moment.duration(endTime.diff(startTime));
+        this.addOptionIfMissing(startOptions, _serializeTime(moment(startTime)));
+        this.addOptionIfMissing(endOptions, _serializeTime(moment(endTime)));
+        this.state = {
+            startTime,
+            endTime,
+            startOptions,
+            endOptions,
+            duration
+        };
+    }
+
+    generateStartTimeOptions = () => {
+        const options = [];
+        const end = moment().endOf('day');
+        const next = moment().startOf('day');
+        let serializedNext;
+        while (next < end) {
+            serializedNext = _serializeTime(moment(next));
+            options.push({key: serializedNext, value: serializedNext, text: serializedNext});
+            next.add(30, 'm');
+        }
+        return options;
+    };
+
+    generateEndTimeOptions = (start) => {
+        const options = [];
+        const end = moment().endOf('day');
+        const next = moment(start).add(30, 'm');
+        let serializedNext, duration, text;
+        while (next < end) {
+            duration = _humanizeDuration(moment.duration(next.diff(start)));
+            serializedNext = _serializeTime(moment(next));
+            text = <div>{serializedNext} <div styleName="duration">({duration})</div></div>;
+            options.push({key: serializedNext, value: serializedNext, text});
+            next.add(30, 'm');
+        }
+        return options;
+    };
+
+    addOptionIfMissing = (options, value) => {
+        const found = options.some((el) => {
+            return el.value === value;
+        });
+        if (!found) {
+            options.unshift({key: value, value, text: value});
+            return true;
+        }
+        return false;
+    };
+
+    updateStartTime = (startTime, endTime, duration) => {
+        const start = moment(startTime, ['HH:mm', 'HHmm', 'Hmm']);
```

Comment:
should we except `HH` as well? so writing just `13` would turn into `13:00`

Reply:
The first formatting also accepts HH